### ISSUE TITLE
More consistent/predictable Frontmatter data usage

### ISSUE THREAD

**Issue by tylersticka** [cloudfour/drizzle-builder#10]
(I understand that some of this may already be in progress, but I wasn't sure if _all_ of it was, hence this issue.)

Frontmatter in Fabricator is weird. Some stuff like `{{notes}}` you edit and access via `notes`. Some stuff like `{{title}}` gets used for some templates but not for others (you can't override the name of a pattern in the nav, for example, which is _crazy_). And anything else gets confined to a `data` object.

I don't really see why an item's Frontmatter can't be the overriding source of truth for that template. At the very least, its properties should be accessible in a consistent way.

**Comment by lyzadanger** [comment]
At this point:

All front-matter data will be available at the top level of the local context object. The local frontmatter will trump (i.e. is later in the `Object.assign` chain) defaults, but a _few_ property-value pairs will take higher precedence (i.e. even later). These high-precedence property-value pairs are things needed internally for the builder to later render and write parsed data/pages/whatever. They are, at present:

* `outputPath`: Full path to where the HTML page will end up
* `resourceType`: `page`, `pattern`, etc. Used by the rendering parser when traversing recursively, to tell the "difference" between different types of entries.

I can make this a little tighter by putting these `drizzle`-needed properties inside their own object. That way there will be only one property that is off-limits for front matter.

tl;dr: use of arbitrary front matter keys is almost unlimited; you'll get all the data in context as you'd expect.

**Comment by erikjung** [comment]
@tylersticka Regarding this..

As it stands, you can now predictably access front matter things.

Within the patterns themselves, the front matter props will be top-level:

```hbs
---
foo: derr
---

<div>
 {{foo}}
</div>
```

When iterating over things with front matter (from another template), the properties will be accessible on the `data` property:

```hbs
{{#each ...}}
  <li>{{data.foo}}</li>
{{/each}}

```
